lighthouse-cli/test/smokehouse/perf/expectations.js

```diff
@@ -65,6 +65,83 @@ module.exports = [
       },
     },
   },
+  {
+    lhr: {
+      requestedUrl: 'http://localhost:10200/perf/perf-budgets/load-things.html',
+      finalUrl: 'http://localhost:10200/perf/perf-budgets/load-things.html',
+      audits: {
+        'resource-summary': {
+          score: null,
+          displayValue: '11 requests • 164 KB',
+          details: {
+            items: [
+              {resourceType: 'total', requestCount: 11, size: '168000±1000'},
+              {resourceType: 'font', requestCount: 2, size: '80000±1000'},
+              {resourceType: 'script', requestCount: 3, size: '55000±1000'},
+              {resourceType: 'image', requestCount: 2, size: '28000±1000'},
+              {resourceType: 'document', requestCount: 1, size: '2100±100'},
+              {resourceType: 'other', requestCount: 2, size: '1250±50'},
+              {resourceType: 'stylesheet', requestCount: 1, size: '450±100'},
+              {resourceType: 'media', requestCount: 0, size: 0},
+              {resourceType: 'third-party', requestCount: 0, size: 0},
+            ],
+          },
+        },
+        'performance-budget': {
+          score: null,
+          details: {
+            items: [
+              {
+                resourceType: 'total',
+                countOverBudget: '3 requests',
+                sizeOverBudget: '65000±1000',
+              },
+              {
+                resourceType: 'script',
+                countOverBudget: '2 requests',
+                sizeOverBudget: '25000±1000',
+              },
+              {
+                resourceType: 'font',
+                countOverBudget: undefined,
+                sizeOverBudget: '4000±500',
+              },
+              {
+                resourceType: 'document',
+                countOverBudget: '1 request',
+                sizeOverBudget: '1100±50',
+              },
+              {
+                resourceType: 'stylesheet',
+                countOverBudget: undefined,
+                sizeOverBudget: '450±100',
+              },
+              {
+                resourceType: 'image',
+                countOverBudget: '1 request',
+                sizeOverBudget: undefined,
```

Comment:
Letting `undefined` values in the expectations match *not defined* properties in `actual` allows testing for things like `sizeOverBudget` when not over budget, [when its value is set to `undefined`](https://github.com/GoogleChrome/lighthouse/blob/ec08e7b9375d3cbe38938b95a0dadfd8287d532e/lighthouse-core/audits/performance-budget.js#L76-L84) and so it lost in the JSON shuffle.

If we just left this `sizeOverBudget` off the expectations it would indeed match an `actual` `undefined` value, which is good...but it would also match if `actual` was erroneously returning a numeric `sizeOverBudget`, so it would be useless.

Reply:
should we add a comment somewhere to this effect?

Reply:
> should we add a comment somewhere to this effect?

added a comment to make it clear to anyone reading the expectations file